Brigitte/sign up

brigittewylee opened the pull request:
Adds the signup page:
<img width="1440" alt="image" src="https://github.com/Your-Missing-Semester/WayPoints/assets/150951722/d98d8e04-b1a6-409f-9785-561d11e2bb4e">

chinkevin25 commented on frontend/src/App.js:
```diff
@@ -1,24 +1,29 @@
-import logo from './logo.svg';
+// import logo from './logo.svg';
```
In general, if we no longer need code, we should just delete!

chinkevin25 commented on frontend/src/App.js:
```diff
@@ -1,24 +1,29 @@
+    // <SignUp>
+    // </SignUp>
+    <LogIn></LogIn>
+    // <div className="App">
+    //   <header className="App-header">
+    //     <img src={logo} className="App-logo" alt="logo" />
+    //     <p>
+    //       Edit <code>src/App.js</code> and save to reload.
+    //     </p>
+    //     <a
+    //       className="App-link"
+    //       href="https://reactjs.org"
+    //       target="_blank"
+    //       rel="noopener noreferrer"
+    //     >
+    //       Learn React
+    //     </a>
+    //   </header>
+    // </div>
```
Let's delete unnecessary code

chinkevin25 commented on frontend/src/App.js:
```diff
@@ -1,24 +1,29 @@
-        </a>
-      </header>
-    </div>
+    // <SignUp>
+    // </SignUp>
+    <LogIn></LogIn>
```
It looks like you did this because you wanted to be able to see what the login page looks like. But eventually we'll going to need both the Login and Signup page to be accessible. The way we would solve something like that is by using something like React Router. 

Let's change this so that react router can set up routing for our pages.

chinkevin25 commented on frontend/src/components/LogIn/gmail_icon.png:
These Oauth icons are duplicated in both the Login and Signup Folders. We should try to avoid duplicating assets if possible. 

Maybe we can create a different folder assets for these two pages to import from.

chinkevin25 commented on frontend/src/components/LogIn/login.css:
```diff
@@ -0,0 +1,120 @@
+    padding-bottom: 10%;
+}
+
+h1 {
```
so selectors like `h1` and `body` are very generic and would end up causing other `h1`s outside of this page to adopt the styling you have defined here.

This can cause unintended styling issues across multiple pages.

Look into https://create-react-app.dev/docs/adding-a-css-modules-stylesheet/ and try to understand what problem CSS Modules is trying to solve! 

In general, what you'd have is something called a "global style sheet" that your entire application should be styled as and then you'd have module style sheets that dictate how this specific page or component should be styled as.

You can imagine a global style sheet might be used to determine present font sizes, or fonts themselves but your module css sheet might dictate positioning or override the global defaults.

chinkevin25 commented on frontend/src/components/LogIn/login.css:
```diff
@@ -0,0 +1,120 @@
+    margin-bottom: 5%;
+}
+
+.loginGroup {
```
by convention, css class names are not camel cased by snake cased. `login-group` small nit, but might be good to adopt the convention!

chinkevin25 commented on frontend/src/components/LogIn/login.css:
```diff
@@ -0,0 +1,120 @@
+    margin-bottom: 3%;
+}
+
+.passReset{
```
I'm noticing some small inconsistencies on the white space in this file

i.e. `input {` vs `passReset{` (note the white space). Lets be consistent in our style and add a white space after the selector for entire file. `passRest {`

chinkevin25 commented on frontend/src/components/SignUp/signup.css:
same comments apply from the other css file

chinkevin25 commented on frontend/src/components/SignUp/signup_error.jsx:
Does this entire file need to exist? The way this file is structure it looks like you're trying to render the entire page if there is an error. That might be work, but it's not the convention nor flexible enough.

What happens if you want to be able to show the error message for the password but not for the email? i.e. email is valid, but the password is not. 

What we're trying to do with the Error states is to conditionally render the error states. Take a look at: https://react.dev/learn/conditional-rendering

What we should do instead, is take the signup page and add some state variables, and update the page state if there is an error message and show that error message if it exists.

chinkevin25 commented on frontend/src/components/SignUp/signup_error.jsx:
```diff
@@ -0,0 +1,39 @@
+            <form>
+                <label>Email</label>
+                    <input className='inputBox' type="text" />
+                    <p className="errorMessage"> Please enter a valid email. </p>
```
Logic wise, how would you determine if the email is valid or not? How should we check?

chinkevin25 commented on frontend/src/components/SignUp/signup_error.jsx:
```diff
@@ -0,0 +1,39 @@
+                    <p className="errorMessage"> Please enter a valid email. </p>
+                <label>Password</label>
+                    <input className='inputBox' type="text" />
+                    <p className="errorMessage"> Passwords must be at least 8 characters
```
Same situation here!

chinkevin25 commented on frontend/src/components/SignUp/signup_error.jsx:
In general too, this file right now is fully "static" there isn't any logic to check if the inputs are correct and there arent any state variables that would be used to have React control the the value of your inputs. (Controlled vs Uncontrolled components):
https://react.dev/learn/sharing-state-between-components#controlled-and-uncontrolled-components


Imagine if we are hit the signup button. How would we collect the values for email and password to send to the server?

chinkevin25 reviewed (changes requested):
Nice first pass! The UI looks great, I like how it looks! I left some comments let me know if you have any questions.

chinkevin25 commented:
You also have some merge conflicts with the main branch so we should resolve this. We can hop on a call together and resolve this together as well just let me know when works for you. But try to resolve the merge conflicts before addressing the comments as Fahad does introduce React Router already so we should try to build on top of his changes.

brigittewylee commented:
**Fixed:**
- Formatting (i.e. whitespace, snake case in css, deleted unused code)
- Changed to css modules
- Removed error page and added conditional rendering components
- Changed login button to link instead of nav 
- Added react router between pages
- Separated oauth logos to own folder

**Updates:**
- Regex for email and password validity

<img width="1418" alt="image" src="https://github.com/Your-Missing-Semester/WayPoints/assets/150951722/407e161f-179f-4a25-adc9-83e4ca5200ad">
